### ISSUE TITLE
gRPC stream lags only on Raydium transactions

### ISSUE THREAD

**Issue by 0xtwoface** [rpcpool/yellowstone-grpc#486]
Hi everyone,

I'm building a real-time backend using Yellowstone and have successfully processed PumpFun transactions, the Raydium price stream via liquidity state, and more. However, I'm encountering an issue when subscribing to the Raydium program **(675kPX9MHTjS2zt1qfr1NYHuzeLXfQM9H24wFSUt1Mp8)** with a simple filter:

```
 const request: SubscribeRequest = {
      accounts: {},
      slots: {},
      transactions: {
        raydiumLiquidityPoolV4: {
          vote: false,
          failed: false,
          signature: undefined,
          accountInclude: [this.RAYDIUM_PROGRAM_ID],
          accountExclude: [],
          accountRequired: [],
        },
      },
      transactionsStatus: {},
      entry: {},
      blocks: {},
      blocksMeta: {},
      accountsDataSlice: [],
      ping: undefined,
      commitment: CommitmentLevel.FINALIZED,
    };
```
  
    
The stream starts lagging by 50-100 slots and the delay continues to increase. This issue only occurs with the Raydium transactions stream, so I'm unsure what might be causing it.

Node Specifications:

CPU: AMD 9354P, 32 Cores @ 3.25 GHz
RAM: 768 GB
Disks: 2 x 480GB NVMe, 2 x 8TB NVMe
Network: 2 x 100Gbps
Egress: 20 TB

I've attempted to adjust the channel capacity with values like 100,000, 250,000, and 1,000,000, but it hasn't resolved the issue.

```
{
  "libpath": "/home/solv/yellowstone-grpc/target/release/libyellowstone_grpc_geyser.so",
  "log": {
    "level": "info"
  },
  "grpc": {
    "address": "0.0.0.0:10000",
    "compression": {
      "accept": [
        "gzip"
      ],
      "send": [
        "gzip"
      ]
    },
    "max_decoding_message_size": "4_194_304",
    "snapshot_plugin_channel_capacity": null,
    "snapshot_client_channel_capacity": "50_000_000",
    "channel_capacity": "1_000_000",
    "unary_concurrency_limit": 100,
    "unary_disabled": false,
    "x_token": "test_rpc"
  },
  "prometheus": {
    "address": "0.0.0.0:8999"
  }
}
```

The CPU load is at 20%, and there are 350 GB of RAM available. Am I missing something, or are the server specifications insufficient to handle subscribing to Raydium transactions?

Any insights or suggestions would be greatly appreciated!

Thank you.

**Comment by fanatid** [comment]
What is the latency between server and consumer? gRPC is lagging on high bandwidth

**Comment by 0xtwoface** [comment]
> What is the latency between server and consumer? gRPC is lagging on high bandwidth

Hi fanatid,

Thank you for your quick response and for pointing that out. It seems that the high latency might be the issue. I hadn't realized it was this significant since it only occurs with this particular stream. I'll try using a different machine with lower latency to see if that resolves the problem.

**Comment by 0xtwoface** [comment]
I just wanted to confirm that the issue was indeed caused by latency. After placing the server and the consumer in the same location, the issue was resolved.

**Comment by fanatid** [comment]
Awesome! Thank you for confirmation!

**Comment by kalloc** [comment]
And the problem is not in latency, my test setup has about 1-2ms to yellowstone endpoint

**Comment by Juanito87** [comment]
Latency wasn't the issue per se in the original post, the latency just added to the issue. If you hit the lag limit, the issue is that you are not ingesting the data fast enough. And the added latency made the problem worse.

**Comment by kalloc** [comment]
10 runs for 60seconds:

Final slot difference: 17 (RPC: 338055516, gRPC: 338055499)
Final slot difference: 70 (RPC: 338055672, gRPC: 338055602)
Final slot difference: 13 (RPC: 338055822, gRPC: 338055809)
Final slot difference: 12 (RPC: 338055975, gRPC: 338055963)
Final slot difference: 103 (RPC: 338056128, gRPC: 338056025)
Final slot difference: 18 (RPC: 338056280, gRPC: 338056262)
Final slot difference: 0 (RPC: 338056432, gRPC: 338056436)
Final slot difference: 4 (RPC: 338056584, gRPC: 338056580)
Final slot difference: 28 (RPC: 338056739, gRPC: 338056711)
Final slot difference: 1 (RPC: 338056891, gRPC: 338056890)

**Comment by kalloc** [comment]
> Latency wasn't the issue per se in the original post, the latency just added to the issue. If you hit the lag limit, the issue is that you are not ingesting the data fast enough. And the added latency made the problem worse.

do you think we need new issue?

**Comment by Juanito87** [comment]
Yes, because the original issue was solved. 
Even if this seems the same, it's doesn't seems to be, as you say there is not ingestion speed nor latency issues.
Please use the template to create the new one.
